Add Windows support for the port server

PatriceVignola opened the pull request:
Using the port server to avoid race conditions was only working on Unix, so this change attemps to extend the support to Windows by using named pipes.

This adds Windows for issue #5

PatriceVignola commented:
@gpshead I'm not sure what's the current state of development for this project, so let me know if it's outside of the scope.

gpshead commented:
While I don't have an ability to directly maintain Windows support, it seems reasonable to have in the module.  It'd be useful if you could also update the Github CI workflows to run tests on Windows.  Otherwise we won't have an easy way to tell if a change might break it.

https://github.com/google/python_portpicker/blob/main/.github/workflows/python-package.yml

PatriceVignola commented:
@gpshead Sure, that makes sense. I'll make the tests pass and update the PR when it's ready.

PatriceVignola commented:
@gpshead I fixed the tests to support Windows and added a Windows pipeline. I tested the pipeline locally, but can we run Github Actions on this PR as a sanity check?

gpshead commented on .github/workflows/python-package.yml:
```diff
@@ -29,3 +35,27 @@ jobs:
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest tox
+          if exist requirements.txt pip install -r requirements.txt
```
I figured out how to get the github windows CI to trigger and committed that config update to your branch.  it's erroring on this line, perhaps it runs under a different shell?

PatriceVignola replied:
Ah, it's using powershell. I pushed an update.

gpshead commented on src/portserver.py:
```diff
@@ -365,8 +423,11 @@ def main():
-    event_loop.remove_signal_handler(signal.SIGUSR1)
+
+    if sys.platform != 'win32':
+        event_loop.run_until_complete(server.wait_closed())
```
Is there a reason this line isn't useful on Windows?  if so, add a comment.

gpshead commented on setup.cfg:
```diff
@@ -33,6 +33,7 @@ platforms = POSIX
 requires =
```
add windows to the supported platforms above here?

gpshead replied:
similarly, increase the version number.  lets go ahead and call this 1.5.0b1

gpshead commented on src/portserver.py:
```diff
@@ -51,11 +53,32 @@ def _get_process_command_line(pid):
+        datetime_result = cmd_result.decode().splitlines()[2].strip()
+
+        try:
+            utc_time = int(datetime_result[-3:])
```
include a reference data format being parsed here in a comment.

gpshead commented on src/portserver.py:
```diff
@@ -51,11 +53,32 @@ def _get_process_command_line(pid):
-        return 0
+    if sys.platform == 'win32':
+        cmd_result = subprocess.run(
+            f'wmic process where ProcessID="{pid}" get CreationDate',
```
Could `int(psutil.Process(pid).create_time())` perhaps do this dance for us?  cross platform even, meaning we wouldn't need our platform specific code...

gpshead reviewed (changes requested):
a couple of minor changes pushed.  otherwise, I believe your logic is correct.  a pile of suggestions, some potentially obsoleting others, that may be able to make this cleaner.

gpshead commented:
One final thing: can you add some 1.5.0 text to ChangeLog.md?
